src/pageServices/contentScripts/highlight.js

```diff
@@ -287,24 +291,39 @@ export const highlightOnPage = () => {
     events.forEach((eventName) => {
       document.removeEventListener(eventName, scrollListenerCallback, true);
     });
-    document.removeEventListener("click", clickListener);
+
+    document.removeEventListener("dblclick", onElementDblClick);
+
     listenersAreSet = false;
   };
 
   const removeHighlight = (callback) => () => {
     removeEventListeners(removeHighlightElements(callback));
   };
 
-  const clickListener = (event) => {
-    if (!event.clientX && !event.clientY) return;
+  const onElementDblClick = (evt) => {
+    const elements = predictedElements?.filter((element) => element.jdnHash === evt.target.id);
```

Comment:
I believe that `find` would be more useful here, hashes are unique and you need only one element.